### PR TITLE
Hide sharing if not logged in, or in an accessible place

### PR DESCRIPTION
**Pull request by davidkelly**
The grand future will have option to login, and so on...

Passing in an array now looks pretty ugly, maybe a one-element array with this data, plus an array of snapshot paths would be better?  I just wanted to make this quickly for now, but happy to ponder a bit more change...

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 47dea0ea2b60ae1f06b940e5cbf30892e99da94f available at: https://gist.github.com/bfbe3c7da0edc26f8909e4511aa6bf2c<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8411-47dea0e.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8411-47dea0e.dmg

File: scripts/system/html/ShareSnapshot.html [hunk 1]
```diff
@@ -28,16 +28,29 @@
...
+    }
     window.onload = function () {
         // Something like the following will allow testing in a browser.
         //addImage({localPath: 'c:/Users/howar/OneDrive/Pictures/hifi-snap-by--on-2016-07-27_12-58-43.jpg'});
         openEventBridge(function () {
             // Set up a handler for receiving the data, and tell the .js we are ready to receive it.
             EventBridge.scriptEventReceived.connect(function (message) {
+                // last element of list contains a bool for whether or not we can share stuff
+                var canShare = message.pop().canShare;
+                print("canShare:"+canShare+", message:" + message.toString())
```
Comment (howard-stearns) [inline scripts/system/html/ShareSnapshot.html]: remove debugging code?

File: scripts/system/html/ShareSnapshot.html [hunk 2]
```diff
@@ -28,16 +28,29 @@
         paths.push(data);
 
     }
+    function handleShareButtons(canShare) {
+        if (!canShare) {
+            // hide the share/do not share buttons
+            document.getElementById("sharing").innerHTML = "<p>You can share if you are logged in and in a public place";
```
Comment (howard-stearns) [inline scripts/system/html/ShareSnapshot.html]: I don't think the first sub-section-header text, "Would you like to share?" makes sense if !canShare. Maybe have it just that the snapshot has been saved? (Maybe only have it change in the !canShare case, or maybe have it always say that. Check with @AlanZimmerman.

Comment (howard-stearns) [inline scripts/system/html/ShareSnapshot.html]: Also, I'm ok with this idea, covering both the login and the shareable case, but again check with Alan as it is different than what we discussed. (See his wording in slack.) If we do break it up into two cases, remember that shareable wins. (Don't make people feel that logging in will do the trick if they're not in a shareable place.)

Comment (howard-stearns) [inline scripts/system/html/ShareSnapshot.html]: Oh wait, I was confused. There's actually a section-header ("Snapshots successfully saved"), which is correct in all cases, and a sub-section-header ("Would you like to share?"), that should be replaced when you cannot share. In other words, the new div you added, with id="sharing", should contain the "Would you like to share?" sub-section-header, in addition to the two buttons.

**Comment by hifi-gustavo** [comment]
Build console output for commit: c1edd008a4f77eeb26fff9f7f95b7dc727d032cc available at: https://gist.github.com/bfbe3c7da0edc26f8909e4511aa6bf2c<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8411-c1edd00.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8411-c1edd00.dmg

**Comment by howard-stearns** [comment]
I tested this in a local build. Pass.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 3408ad6f545e24ec35a289d8303d1bccc93121dc available at: https://gist.github.com/bfbe3c7da0edc26f8909e4511aa6bf2c<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8411-3408ad6.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8411-3408ad6.dmg

**Pull request merged**